prime-router/src/main/kotlin/fhirengine/engine/FHIRTranslator.kt

```diff
@@ -70,56 +68,51 @@ class FHIRTranslator(
         val provenance = bundle.entry.first { it.resource.resourceType.name == "Provenance" }.resource as Provenance
         val receiverEndpoints = provenance.target.map { it.resource }.filterIsInstance<Endpoint>()
 
-        receiverEndpoints.forEach { receiverEndpoint ->
-            val receiverName = receiverEndpoint.identifier[0].value
-            val receiver = settings.findReceiver(receiverName)
+        val receiverAndEndpoints: List<Pair<Endpoint, Receiver>> = receiverEndpoints.mapNotNull { endpoint ->
```

Comment:
This code is equivalent I've just cleaned it up to be easier understand what is being looked for